fdbclient/SpecialKeySpace.actor.h

```diff
@@ -38,13 +38,55 @@ class SpecialKeyRangeBaseImpl {
 	// Each derived class only needs to implement this simple version of getRange
 	virtual Future<Standalone<RangeResultRef>> getRange(ReadYourWritesTransaction* ryw, KeyRangeRef kr) const = 0;
 
-	explicit SpecialKeyRangeBaseImpl(KeyRangeRef kr) : range(kr) {}
+	explicit SpecialKeyRangeBaseImpl(KeyRangeRef kr, bool async = false) : range(kr), async(async) {}
 	KeyRangeRef getKeyRange() const { return range; }
+	bool isAsync() const { return async; }
```

Comment:
FWIW `isAsync` doesn't seem very descriptive to me. The key piece of information to convey IMO is that we can only call getRange once. I don't (yet?) have a better suggestion for a name though

Reply:
Yeah,  something more descriptive is needed here. I add a `TODO` here, and we can find a good name later when all other comments and tests are done.